### PR TITLE
config.h can't be prefixed unconditionally

### PR DESCRIPTION
**Pull request by pprindeville**
The script `make-event-config.sed` was mangling all the symbols by prefixing them with `EVENT__`.  The problem here is that some symbols aren't for local consumption within libevent, but rather influence other system header files (ex: `__USE_FILE_OFFSET64` is used by dozens of header files including `<sys/sendfile.h>`).

As a workaround, all symbols starting with a capital letter only (with the exception of `STDC_HEADERS` which must also be left untouched) will be mangled.

Future contributors will need to be aware of this distinction.

### CODE REVIEW

**Comment by pprindeville** [comment]
Not sure how to handle cases of `EVENT___func__` or `EVENT___cplusplus` in the source code.

File: CMakeLists.txt [hunk 1]
```diff
@@ -515,10 +515,6 @@ if (EVENT__HAVE_KQUEUE)
     endif()
 endif()
 
-CHECK_SYMBOL_EXISTS(_MINIX "stdio.h" EVENT___MINIX)
-CHECK_SYMBOL_EXISTS(_POSIX_1_SOURCE "stdio.h" EVENT___POSIX_1_SOURCE)
-CHECK_SYMBOL_EXISTS(_POSIX_SOURCE "stdio.h" EVENT___POSIX_SOURCE)
-
```
Comment (azat) [inline CMakeLists.txt]: and why did you remove those, because they are unused?

File: WIN32-Code/nmake/event2/event-config.h [hunk 2]
```diff
@@ -341,24 +341,24 @@
...
 /* Define to `unsigned' if <sys/types.h> does not define. */
-/* #undef EVENT__size_t */
+/* #undef size_t */
 
 /* Define to unsigned int if you dont have it */
-#define EVENT__socklen_t unsigned int
+#define socklen_t unsigned int
 
 /* Define to `int' if <sys/types.h> does not define. */
-#define EVENT__ssize_t SSIZE_T
```
Comment (azat) [inline WIN32-Code/nmake/event2/event-config.h]: have you tried to compile libevent with this header on win32?

File: event-config.h.cmake [hunk 3]
```diff
@@ -473,15 +473,6 @@
...
 
-/* set an alias for whatever __func__ __FUNCTION__ is, what sillyness */
-#if defined (__func__)
-#define EVENT____func__ __func__
-#elif defined(__FUNCTION__)
-#define EVENT____func__  __FUNCTION__
-#else
-#define EVENT____func__ __FILE__
-#endif
-
```
Comment (azat) [inline event-config.h.cmake]: hm, I'm looking at a46c2609bfa4d68e78665dd55461af9b1ce9651b and it looks to me that `__func__` were introduced with a purpose, have you seen it?

File: CMakeLists.txt (hunk 1 above)
Comment (pprindeville) [reply CMakeLists.txt]: My understanding is that the symbols get emitted unmodified by default, but maybe I'm wrong.  I don't use `cmake`.

File: WIN32-Code/nmake/event2/event-config.h (hunk 2 above)
Comment (pprindeville) [reply WIN32-Code/nmake/event2/event-config.h]: No, I don't have access to a Windows development environment.  Doesn't Travis build with AppVeyor?

File: CMakeLists.txt (hunk 1 above)
Comment (pprindeville) [reply CMakeLists.txt]: Correct.

Comment (azat) [reply CMakeLists.txt]: All I'm saying is that this is cleanup, and we shouldn't mix **fixing real issues** with **cleanups**

File: WIN32-Code/nmake/event2/event-config.h (hunk 2 above)
Comment (azat) [reply WIN32-Code/nmake/event2/event-config.h]: Yes it does, but not sure that it uses this pre-generated header. Personally I, would prefer to leave only `cmake` builds, this will fix a lot of caveats, but I guess that we can't do this right now.

I have some vagrant boxes (that includes windows), I hadn't used them for a few months, so it will took some time.

File: event-config.h.cmake (hunk 3 above)
Comment (azat) [reply event-config.h.cmake]: AFAICS that patch mentioned why they were introduced (I guess some compiler hadn't implemented `__func__`), but since IRIX is EOL, I think we should clean this up (but not only this two places but also `configure.ac` and `CMakeLists.txt`), and we should check win32 case, since IIRC in some versions of toolchain there is only `__FUNCTION__` and no `__func__`

Comment (azat) [reply event-config.h.cmake]: But let's keep it separate from this issue anyway (at least separate patch in this series)

**Comment by azat** [comment]
> CB is failing for Mac but it's not clear why... it mostly looks like some tests are taking slightly longer to run than expected.

Not sure what CB means, but if we are talking about travis-ci there was some problems on macos.

**Comment by pprindeville** [comment]
Any movement on this?

**Comment by azat** [comment]
I just applied the first patch in the series.

As for the second patch, I still haven't restored my vagrant playground
for this (to check all caveats in other OS'es), apparently it will take
some more time...

File: CMakeLists.txt (hunk 1 above)
Comment (pprindeville) [reply CMakeLists.txt]: Okay, so drop these symbols as part of a separate PR or a separate commit?

File: event-config.h.cmake (hunk 3 above)
Comment (pprindeville) [reply event-config.h.cmake]: See PR #644

File: CMakeLists.txt (hunk 1 above)
Comment (pprindeville) [reply CMakeLists.txt]: See PR #645.

**Comment by pprindeville** [comment]
Closing and splitting into other PR's.
